python/ray/experimental/data/tests/test_dataset.py

```diff
@@ -1097,6 +1097,57 @@ def test_csv_write(ray_start_regular_shared, tmp_path):
     shutil.rmtree(path)
 
 
+def test_sort_simple(ray_start_regular_shared):
+    num_items = 100
+    parallelism = 4
+    xs = list(range(num_items))
+    random.shuffle(xs)
+    ds = ray.experimental.data.from_items(xs, parallelism=parallelism)
+    assert ds.sort().take(num_items) == list(range(num_items))
+    assert ds.sort(descending=True).take(num_items) == list(
+        reversed(range(num_items)))
+    assert ds.sort(key=lambda x: -x).take(num_items) == list(
+        reversed(range(num_items)))
+
+
+def test_sort_arrow(ray_start_regular_shared):
+    _test_sort_arrow()
+
+
+def test_sort_arrow_distributed(ray_start_regular_shared):
+    _test_sort_arrow(parallelism=4)
+
+
+def _test_sort_arrow(num_items=100, parallelism=1):
+    a = list(reversed(range(num_items)))
+    b = [f"{x:03}" for x in range(num_items)]
+    shard = int(np.ceil(num_items / parallelism))
+    offset = 0
+    dfs = []
+    while offset < num_items:
+        dfs.append(
+            pd.DataFrame({
+                "a": a[offset:offset + shard],
+                "b": b[offset:offset + shard]
+            }))
+        offset += shard
+    if offset < num_items:
+        dfs.append(pd.DataFrame({"a": a[offset:], "b": b[offset:]}))
+    ds = ray.experimental.data.from_pandas([ray.put(df) for df in dfs])
+
+    def assert_sorted(sorted_ds, expected_rows):
+        assert [tuple(row.values())
+                for row in sorted_ds.iter_rows()] == list(expected_rows)
+
+    assert_sorted(ds.sort(key="a"), zip(reversed(a), reversed(b)))
+    assert_sorted(ds.sort(key="b"), zip(a, b))
+    assert_sorted(ds.sort(key="a", descending=True), zip(a, b))
+    assert_sorted(
+        ds.sort(key=[("b", "descending")]), zip(reversed(a), reversed(b)))
+
+
 if __name__ == "__main__":
     import sys
-    sys.exit(pytest.main(["-v", __file__]))
+    # sys.exit(pytest.main(["-v", __file__]))
+    sys.exit(
+        pytest.main(["-sv", __file__, "-k", "test_sort"]))
```

Comment:
Remove and use the commented out command instead?

Reply:
oops will do